optuna/pruners/hyperband.py

```diff
@@ -122,9 +135,50 @@ def __init__(
             )
             warnings.warn(message, DeprecationWarning)
             _logger.warning(message)
-            self._n_brackets = n_brackets
 
-        self._trial_allocation_budgets = []  # type: List[int]
+        if min_early_stopping_rate_low is not None:
+            message = (
+                "The argument of `min_early_stopping_rate_low` is deprecated. "
+                "Please specify `min_resource` appropriately."
+            )
+            warnings.warn(message, DeprecationWarning)
+            _logger.warning(message)
+
+    def prune(self, study: "optuna.study.Study", trial: FrozenTrial) -> bool:
+        if len(self._pruners) == 0:
+            self._try_initialization(study)
+            if len(self._pruners) == 0:
+                return False
+
+        i = self._get_bracket_id(study, trial)
+        _logger.debug("{}th bracket is selected".format(i))
+        bracket_study = self._create_bracket_study(study, i)
+        return self._pruners[i].prune(bracket_study, trial)
+
+    def _try_initialization(self, study: "optuna.study.Study") -> None:
+        if self._max_resource == "auto":
```

Comment:
Not a strong opinion as `n_brackets` is deprecated but you only have to run the code inside this `if` when `n_brackets` is not given. It'd better keep compatibility for current users.

Reply:
Thank you for your pointing. Actually, we waste some trials used to determine `max_resource` when `max_resource="auto"` and `n_brackets` is given. I think it does not matter because `n_brackets` will be removed. So it's ok as it is.

Reply:
Alright, that is okay as well.